server.py

```diff
@@ -146,6 +143,20 @@ def dashboard():
     contests = Contest.select().order_by(Contest.start_time)
     return bottle.template("dashboard.html", contests=contests)
 
+@app.get("/stats")
+@login_required
+def statistics():
+    sub_history_temp = Submission.select(Contest.code, ContestProblems.question, Submission.time, Submission.is_correct)\
+        .where(Submission.user == Session.get(Session.token == bottle.request.get_cookie("s_id")).user)\
+        .join(ContestProblems, on=(Submission.contestProblem == ContestProblems.id)) \
+        .join(Session, on=(Submission.user == Session.user))\
+        .switch() \
```

Comment:
@Raul9595 You don't need to join the Session table anymore, the where statement should be enough.

Reply:
We can even avoid joining `contest`.
Refer: https://github.com/PyJaipur/PyJudge/pull/126#discussion_r285917308
Using `backref` and with the foreign key, you can directly refer to `contest` from `contestProblems`.
But wait for @theSage21 approval for this one, he might have better suggestions.

Reply:
you could. I think this query might be more clear. I tend to use the backref feature sparingly since it introduces db requests without you realizing it at first glance.

Reply:
I have not used backref against the recommendation of @theSage21. I have removed the Session table join though.